contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+     *      stablecoin via Uniswap
+     */
+    function harvestAndSwap() external nonReentrant {
+        // TODO: add protection so that harvestAndSwap isn't called twice too closely together
+        _harvest();
+        _swap();
+        IVault(vaultAddress).rebase();
+    }
+
+    /**
+     * @dev Collect reward tokens from all strategies
+     */
+    function _harvest() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+        for (uint256 i = 0; i < allStrategies.length; i++) {
+            _harvest(allStrategies[i]);
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvest(address _strategyAddr) external onlyGovernor nonReentrant {
+        // TODO: add protection so that harvestAndSwap isn't called twice too closely together
+        _harvest(_strategyAddr);
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy and swap for supported
+     *      stablecoin via Uniswap. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvestAndSwap(address _strategyAddr) external nonReentrant {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        _harvest(address(strategy));
+        address[] memory rewardTokens = strategy.getRewardTokenAddresses();
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            _swap(rewardTokens[i]);
+        }
+        IVault(vaultAddress).rebase();
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and swap them for a
+     *      supported stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from.
+     */
+    function _harvest(address _strategyAddr) internal {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        address[] memory rewardTokenAddresses = strategy
+            .getRewardTokenAddresses();
+
+        if (rewardTokenAddresses.length > 0) {
+            strategy.collectRewardTokens();
+        }
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function _swap() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            _swap(swapTokens[i]);
+        }
+    }
+
+    /**
+     * @dev Swap a record token for stablecoins for Uniswap. The token must have
+     *       a registered price feed with the price provider.
+     * @param _swapToken Address of the token to swap.
+     */
+    function _swap(address _swapToken)
+        internal
+        returns (uint256[] memory swapResult)
+    {
+        RewardTokenConfig storage tokenConfig = rewardTokenConfigs[_swapToken];
```

Comment:
We probably want to load this to `memory` here, rather than reloading from `storage` every time we get a value from it. 


Would be worth comparing the gas usage difference between memory and storage in this line.

Reply:
Thanks you are right. Ran a gas estimate function and calling `harvestAndSwap` total gas usage reduced from 466115 to 465213 (reduced by 902) when changing from `storage` to `memory`.